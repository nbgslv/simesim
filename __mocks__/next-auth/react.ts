const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
