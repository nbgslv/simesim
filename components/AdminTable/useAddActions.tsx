import React, {useMemo} from "react";
import RowActions from "./RowActions";
import {
    GridColumns, GridRowModel,
    GridRowModesModel,
    GridValidRowModel
} from "@mui/x-data-grid";

type RowActionsProps<T extends GridValidRowModel> = {
    columns: GridColumns;
    rowModesModel: GridRowModesModel,
    rowsActions: string[]
    setRowModesModel: React.Dispatch<React.SetStateAction<GridRowModesModel>>
    setRows: React.Dispatch<React.SetStateAction<GridRowModel<T>[]>>
    rows: GridRowModel<T>[]
}

const useAddActions = <T extends GridValidRowModel>({ columns, rowsActions, rowModesModel, setRowModesModel, setRows, rows }: RowActionsProps<T>) => {
    const _columns = useMemo(
        () =>
            columns.concat({
                field: 'actions',
                type: 'actions',
                headerName: 'Actions',
                getActions: (row) =>
                    [
                        <>
                            {/* @ts-ignore*/}
                            <RowActions
                                rowsActions={rowsActions}
                                row={row as unknown as GridRowModel<T>}
                                rowModesModel={rowModesModel}
                                setRowModesModel={setRowModesModel}
                                setRows={setRows}
                                rows={rows}
                            />
                        </>
                    ]
            }),
        [columns, rowsActions, rowModesModel, setRowModesModel, setRows, rows]
    );

    return { columns: _columns };
}

export default useAddActions;
