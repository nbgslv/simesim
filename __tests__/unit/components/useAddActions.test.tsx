import { renderHook } from '@testing-library/react';
import { GridRowModesModel } from '@mui/x-data-grid';
import useAddActions from '../../../components/AdminTable/useAddActions';

describe('useAddActions', () => {
  const columns = [
    { field: 'id', headerName: 'ID', width: 100 },
    { field: 'name', headerName: 'Name', width: 200 },
  ];
  const rowsActions = ['delete'];
  const rowModesModel = { '1': { mode: 'view' }, '2': { mode: 'view' } };
  const setRowModesModel = jest.fn();
  const setRows = jest.fn();
  const rows = [
    { id: 1, name: 'John Doe' },
    { id: 2, name: 'Jane Doe' },
  ];
  const onDelete = jest.fn(() => Promise.resolve(true));

  beforeEach(() => {
    setRowModesModel.mockClear();
    setRows.mockClear();
  });

  it('should add an actions column to the columns array', () => {
    const { result } = renderHook(() =>
      useAddActions({
        columns,
        rowsActions,
        rowModesModel: rowModesModel as GridRowModesModel,
        setRowModesModel,
        setRows,
        rows,
        onDelete,
      })
    );
    const newColumns = result.current.columns;
    expect(newColumns).toHaveLength(3);
    expect(newColumns[2].field).toEqual('actions');
  });
});
