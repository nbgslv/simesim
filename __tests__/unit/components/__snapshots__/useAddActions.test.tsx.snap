// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useAddActions renders correctly 1`] = `
{
  "current": {
    "columns": [
      {
        "field": "id",
        "headerName": "ID",
        "width": 100,
      },
      {
        "field": "name",
        "headerName": "Name",
        "width": 200,
      },
      {
        "field": "actions",
        "getActions": [Function],
        "headerName": "Actions",
        "type": "actions",
      },
    ],
  },
}
`;
